cmd/woodpecker-autoscaler/flags.go

```diff
@@ -12,95 +12,99 @@ var flags = []cli.Flag{
 		Name:    "log-level",
 		Value:   "info",
 		Usage:   "default log level",
-		EnvVars: []string{"WOODPECKER_LOG_LEVEL"},
+		Sources: cli.EnvVars("WOODPECKER_LOG_LEVEL"),
 	},
 	&cli.StringFlag{
 		Name:    "reconciliation-interval",
 		Value:   "1m",
 		Usage:   "interval at which the autoscaler will reconcile as duration string like 2h45m (https://pkg.go.dev/time#ParseDuration)",
-		EnvVars: []string{"WOODPECKER_RECONCILIATION_INTERVAL"},
+		Sources: cli.EnvVars("WOODPECKER_RECONCILIATION_INTERVAL"),
 	},
 	&cli.StringFlag{
 		Name:    "pool-id",
 		Value:   "1",
 		Usage:   "id of the autoscaler pool",
-		EnvVars: []string{"WOODPECKER_POOL_ID"},
+		Sources: cli.EnvVars("WOODPECKER_POOL_ID"),
 	},
 	&cli.IntFlag{
 		Name:    "min-agents",
 		Value:   1,
 		Usage:   "minimum amount of agents",
-		EnvVars: []string{"WOODPECKER_MIN_AGENTS"},
+		Sources: cli.EnvVars("WOODPECKER_MIN_AGENTS"),
 	},
 	&cli.IntFlag{
 		Name:    "max-agents",
 		Value:   10,
 		Usage:   "maximum amount of agents",
-		EnvVars: []string{"WOODPECKER_MAX_AGENTS"},
+		Sources: cli.EnvVars("WOODPECKER_MAX_AGENTS"),
 	},
 	&cli.StringFlag{
 		Name:    "agent-inactivity-timeout",
 		Value:   "10m",
 		Usage:   "time an agent is allowed to be inactive before it can be terminated as duration string like 2h45m (https://pkg.go.dev/time#ParseDuration)",
-		EnvVars: []string{"WOODPECKER_AGENT_INACTIVITY_TIMEOUT", "WOODPECKER_AGENT_ALLOWED_STARTUP_TIME"},
+		Sources: cli.EnvVars("WOODPECKER_AGENT_INACTIVITY_TIMEOUT", "WOODPECKER_AGENT_ALLOWED_STARTUP_TIME"),
 	},
 	&cli.StringFlag{
 		Name:    "agent-idle-timeout",
 		Value:   "10m",
 		Usage:   "time an agent is allowed to be idle before it can be terminated as duration string like 2h45m (https://pkg.go.dev/time#ParseDuration)",
-		EnvVars: []string{"WOODPECKER_AGENT_IDLE_TIMEOUT"},
+		Sources: cli.EnvVars("WOODPECKER_AGENT_IDLE_TIMEOUT"),
 	},
 	&cli.IntFlag{
 		Name:    "workflows-per-agent",
 		Value:   2,
 		Usage:   "max workflows an agent will executed in parallel",
-		EnvVars: []string{"WOODPECKER_WORKFLOWS_PER_AGENT"},
+		Sources: cli.EnvVars("WOODPECKER_WORKFLOWS_PER_AGENT"),
 	},
 	&cli.StringFlag{
 		Name:    "server-url",
 		Value:   "http://localhost:8000",
 		Usage:   "woodpecker server address",
-		EnvVars: []string{"WOODPECKER_SERVER"},
+		Sources: cli.EnvVars("WOODPECKER_SERVER"),
 	},
 	&cli.StringFlag{
-		Name:     "server-token",
-		Usage:    "woodpecker api token",
-		EnvVars:  []string{"WOODPECKER_TOKEN"},
-		FilePath: os.Getenv("WOODPECKER_TOKEN_FILE"),
+		Name:  "server-token",
+		Usage: "woodpecker api token",
+		Sources: cli.ValueSourceChain{
```

Comment:
What's the purpose of this chain here. Is this the new default for multiple sources?

Reply:
Not sure what you mean with "new default". It's afaik the only way in v3 to specify multiple input sources (`EnvVars` and `FilePath`)

Reply:
```suggestion
		Sources: cli.NewValueSourceChain(
			cli.EnvVar("WOODPECKER_HETZNERCLOUD_USERDATA"),
			cli.File(os.Getenv("WOODPECKER_HETZNERCLOUD_USERDATA_FILE")),
		),
```

Reply:
Do you prefer this way? Then I need to change it in other locations as well

Reply:
As I don't care that much, I switched to `NewValueSourceChain` now.